src/Sulu/Component/Rest/RestHelper.php

```diff
@@ -63,9 +68,22 @@ public function initializeListBuilder(ListBuilderInterface $listBuilder, array $
             $listBuilder->search($this->listRestHelper->getSearchPattern());
         }
 
+        // add sort
         $sortBy = $this->listRestHelper->getSortColumn();
         if ($sortBy != null) {
             $listBuilder->sort($fieldDescriptors[$sortBy], $this->listRestHelper->getSortOrder());
         }
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function addFilters(ListBuilderInterface $listBuilder, array $fieldDescriptors)
+    {
+        foreach ($this->listRestHelper->getFilters() as $filterKey => $filterValue) {
+            if (isset($fieldDescriptors[$filterKey])) {
```

Comment:
This partly solves the issue above, but it will still result in some unexpected behavior when a controller adds a parameter which also exists as a field descriptor.